### PR TITLE
WIP: Makes dotnet-install.ps1 x-platform

### PR DESCRIPTION
**Pull request by mklement0**
Adds cross-edition and cross-platform support to the script, addresses #8278.

### CODE REVIEW

**Comment by nguerrera** [comment]
There appear to be a lot of style changes unrelated to the task at hand. Examples: renaming Say-Verbose to SayVerbose, Double quotes to single quotes, etc.

That makes it very hard to review. Since this script rolls out from here to live production environments, we need to assess the risk of any change very carefully. Please propose style changes independently from functional changes.

**Comment by mklement0** [comment]
Re style changes: 

Under normal circumstances, I fully agree that such change should be submitted separately, but given that this PR - of necessity - amounts to a substantial rewrite, I was hoping we can just leave them in.

One aspect of the rewrite was to make the script more PowerShell-idiomatic, particularly with a view to not generating warnings when editing the code in Visual Studio Code with the PowerShell extension.

* `Say-Verbose` was renamed to `SayVerbose`, because `Say` is not an approved verb for naming cmdlets / advanced functions.

* Double quotes were replaced with single quotes in all cases where the string's content is effectively taken _literally_, i.e., where no interpolation (expansion) is needed. This is good practice and facilitates (future) readability, because you can tell at a glance whether a given string will be used as-is or is subject to expansion.

> Since this script rolls out from here to live production environments, we need to assess the risk of any change very carefully. 

Again, fully agreed - but given the substantial changes in this PR, it is _comprehensive tests_ that need to ensure that.

The PR is currently lacking such tests, because - after having spent a substantial amount authoring this PR - _I still had no assurance that there was a fundamental willingness to accept a PR of this nature due to its scope_ 

In short:

* If we agree that this PR - once eventually completed to everyone's satisfaction - is worthwhile, including maintaining the cross-platform nature of the script going forward, at least in the foreseeable future _in parallel_ with the Bash script...

* ... and someone helps me get started with authoring the requisite tests

I'm happy to continue work on this PR.

Honestly, I don't want to spend the time factoring out the incidental changes, and just letting this die is a possible - if unfortunate - outcome too.

**Comment by nguerrera** [comment]
Thanks @mklement0. I full agree on waiting for consensus that we will take the change before doing work to factor changes out. Also, I agree we should have tests. But if when/we have that, I will still ask for things that make the script more idiomatic or remove warnings from powershell, but do not functionally contribute to the x-platform goal separate. I support making those changes, but I want to review them separately is all.

**Comment by mklement0** [comment]
Thanks, @nguerrera.

>  I fully agree on waiting for consensus

We can cross that bridge first and take it from there.

But - what is the process for reaching this consensus?

I've made the case for this PR in https://github.com/dotnet/cli/issues/8278#issuecomment-457931158, but aside from @SeeminglyScience's gracious show of support via an emoji, there has been no feedback.

**Comment by amis92** [comment]
Why is this dead. It's such a great contribution. :(

**Comment by mklement0** [comment]
Thanks, @amis92; I can't speak to why there's been no decision on whether to fundamentally accept such a contribution (though perhaps the fact the maintenance burden would increase as long as PowerShell isn't available by default everywhere  plays a part).

If you're feeling adventurous, you can grab a copy of the script from [here](https://raw.githubusercontent.com/mklement0/cli/dotnet-install-ps1-xplat/scripts/obtain/dotnet-install.ps1) - I've just synced it with the latest official scripts.

I've used it personally without problems, but mostly in vanilla scenarios, and by no means on all supported platforms, and without formal testing.

**Comment by sfoslund** [comment]
@mklement0 sorry for leaving this for so long. The install scripts have been moved to [their own repo](https://github.com/dotnet/install-scripts) so I'm closing this but feel free to send a PR there instead and the team can take a look. Thanks!
